### PR TITLE
fix: add explicit node-fetch dependency

### LINKED ISSUE #1260: ERR_REQUIRE_ESM - node-fetch 3.x issue
**Describe the bug**

When running redocly-cli with an installed node-fetch@3.x we get the error:

```
> redocly build-docs swagger-spec.json --output=public/nd-swagger-fe-docs.html

/Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/lib/fetch-with-timeout.js:12
const node_fetch_1 = require("node-fetch");
                     ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/node-fetch/src/index.js from /Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/lib/fetch-with-timeout.js not supported.
Instead change the require of index.js in /Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/lib/fetch-with-timeout.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/lib/fetch-with-timeout.js:12:22)
    at Object.<anonymous> (/Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/lib/utils.js:24:30)
    at Object.<anonymous> (/Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/lib/assert-node-version.js:4:17)
    at Object.<anonymous> (/Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/lib/index.js:13:1)
    at Object.<anonymous> (/Users/joantune/SourceCode/SoundsGood/nd/nd-sga-backend/node_modules/@redocly/cli/bin/cli.js:3:1) {
  code: 'ERR_REQUIRE_ESM'
}
```

**To Reproduce**

install latest node-fetch (defaults to 3.x ATM)
install redocly-cli
run redocly-cli with parameters `redocly build-docs swagger-spec.json --output=public/nd-swagger-fe-docs.html` (unsure if parameters are required)

**Expected behavior**

Command runs :)

**Logs**

Already above

**Redocly Version(s)**

1.0.2

**`Node.js` Version(s)**

v16.16.0

**Workaround**

Downgrade node-fetch to 2.x :/ 

** Suggested Solution **

Use import instead of require on file seen above in the desc

### PR DESCRIPTION
**Pull request by billytrend-cohere**
## What/Why/How?

Fixes #1260 

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows (internal)
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by tatomyr** [comment]
Hi! Could you add a patch [changeset](https://github.com/Redocly/redocly-cli/blob/main/.github/CONTRIBUTING.md#pull-request-guidelines)?

**Comment by billytrend-cohere** [comment]
Done thanks!

File: .changeset/nice-houses-hug.md
Comment (tatomyr) [inline .changeset/nice-houses-hug.md:5]:
```suggestion
Fixed an issue with resolving node-fetch dependency.
```

File: .changeset/nice-houses-hug.md
Comment (tatomyr) [inline .changeset/nice-houses-hug.md:5]:
```suggestion
Fixed an issue with resolving the `node-fetch` package by explicitly adding the missing dependency.
```

Comment (tatomyr) [reply .changeset/nice-houses-hug.md]: This only solves an installation error. Updated the description.

**Review by lornajane — APPROVED** [review]
Thanks @billytrend-cohere !

**Comment by billytrend-cohere** [comment]
Thanks all! when do we expect a new version to be released? would unblock me :)

**Comment by tatomyr** [comment]
Thank you for contributing, @billytrend-cohere 🎉  
I'd like to make the release today. Waiting for approvals.

**Comment by tatomyr** [comment]
Released

**Pull request merged**
